src/swarm/neo/client/Connection.d

```diff
@@ -462,8 +462,20 @@ public final class Connection: ConnectionBase
         if (!this.restart_after_shutdown)
             return false;
 
-        this.status_ = this.status_.Connecting;
-        retry(this.tryConnect(), this.send_loop, this.epoll);
+        try
+        {
+            this.status_ = this.status_.Connecting;
+            retry(this.tryConnect(), this.send_loop, this.epoll);
+        }
+        catch (TimerException e)
+        {
+            // The timer failed - This can happen only due to OS resource
+            // exhaustion.
+            Log.lookup(this.classinfo.name).error("connect: The retry " ~
```

Comment:
It's a bit of a problem doing the log lookup in this case, since that's likely to fail. I suggest having a static module logger instead.

Reply:
(to be more precise - it is likely to fail, since it will try open a new file, if not already open).